clients/unshaded/src/main/java/tachyon/client/block/RemoteBlockInStream.java

```diff
@@ -30,6 +33,13 @@
 public final class RemoteBlockInStream extends BufferedBlockInStream {
   /** The address of the worker to read the data from. */
   private final InetSocketAddress mLocation;
+  /** The returned lock id after acquiring the block lock. */
+  private final Long mLockId;
+
+  /** Client to communicate with the remote worker. */
+  private final WorkerClient mWorkerClient;
```

Comment:
I think we have discussed about on-demand acquire/release on this PR before, it might not be an option, lock/unlock has to be called by the same client.